src/lib.rs

```diff
@@ -21,28 +22,39 @@ use crate::keys_manager::{generate_secret, init_keys_manager};
 use crate::logger::LightningLogger;
 use crate::secret::Secret;
 use crate::storage_persister::StoragePersister;
+use crate::tx_broadcaster::TxBroadcaster;
 
 use bitcoin::Network;
+use esplora_client::r#async::AsyncClient;
+use esplora_client::Builder;
 use lightning::util::config::UserConfig;
 use log::{info, warn, Level as LogLevel};
+use std::sync::Arc;
 
+#[allow(dead_code)]
 pub struct LightningNode {
-    rt: AsyncRuntime,
+    rt: Arc<AsyncRuntime>,
+    esplora_client: Arc<AsyncClient>,
 }
 
 impl LightningNode {
     pub fn new(
         config: Config,
         redundant_storage_callback: Box<dyn RedundantStorageCallback>,
     ) -> Result<Self, InitializationError> {
-        let rt = AsyncRuntime::new()?;
+        let rt = Arc::new(AsyncRuntime::new()?);
```

Comment:
Instead of wrapping the runtime in an `Arc`, I think we can use a `tokio::runtime::Handle`.

Reply:
You are right, I will address it in a separate PR.

Reply:
https://github.com/getlipa/lipa-lightning-lib/pull/33